docs/testing.md

```diff
@@ -0,0 +1,84 @@
+
+Testing libraries:
+
+1. Make sure you are up to date with std lib [testing](https://pkg.go.dev/testing) functionality:
+   - Use `FuzzyXXX` tests to generate scenarios
+   - Use `t.Skip` to skip long or not stable tests.
+     **NOTE**: We should always run all tests before release. So the idea is that long tests won't impact daily CI, but we can make rule that this will be run only once a day and before release.
+1. Use [gotest.tools](https://pkg.go.dev/gotest.tools/assert) instead of `stretchr/testify/suite`.
+1. Use [gomock](github.com/golang/mock/mockgen) to generate mock interfaces.
+1. Use [rapid](https://github.com/flyingmutant/rapid) for fuzzy testing (especially in unit tests).
+1. Use module fixtures. Example in [x/leverage/fixtures](../x/leverage/fixtures)
+
+Our CI should use all tests (unit, integration, e2e) except those which are very long or not stable.
+Long tests should be used periodically (eg once every second day) instead, to not significatly decrease the pull request experience.
+
+Finally all tests (including QA) must be run on release tag (including `beta` and `rc`).
+
+## Unit Tests
+
+Unit testing is a software testing method by which individual units of source code—sets of one or more computer program modules together with associated control data, usage procedures, and operating procedures—are tested to determine whether they are fit for use.
+
+Unit tests should not have external dependencies (or mock them) except the function or object under the test.
+
+Mock implementations should be provided for all dependencies.
+
+## Integration Tests
+
+Tests which involve at least 2 dependencies tested together. Usually they test integration between 2 or more objects in their expected setup.
+
+Ideally we can test integration between local components without involving the whole app.
+
+At Umee our Keeper tests are usually app level _integration tests_ -- this means that usually we need to setup the whole app to execute the tests, limiting the control of objects under the tests. Our goal should be rather to be able to test modules without setting up an app.
+
+## Simulation Tests
+
+Simulations are very high level App integration tests, where:
+
+- we only run App in it's expected setup (without the normally expected processes, like price-feeder)
+- App is producing blocks on the chain until a specified height is reached.
+  at each block the sim test runner sequentially ask each module to generate random transactions (represented as operations) and execute them directly in the app transaction runner.
+- during the sim runner, we don't execute any user defined assertions. Instead we expect the chain to function correctly (don't return errors nor panics). Hence, when creating a new operation (random sim transaction), we need to check conditions to satisfy that a transaction won't be rejected (like checking bank balance etc...).
+
```

Comment:
Adding this link under the simulation section could be helpful: https://pkg.go.dev/github.com/cosmos/cosmos-sdk/x/simulation#SimulateFromSeed

Reply:
added